### ISSUE TITLE
Assumptions for zoo is_real, is_positive etc

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#16415]
Currently on master we have:
```julia
In [1]: zoo.is_real                                                                                                                            
Out[1]: False

In [2]: zoo.is_positive                                                                                                                        
Out[2]: False

In [3]: zoo.is_negative                                                                                                                        
Out[3]: False
```
However:
```
In [4]: oo.is_real                                                                                                                             
Out[4]: True

In [5]: oo.is_positive                                                                                                                         
Out[5]: True
```

My interpretation of zoo is that it is infinite with uncertain complex phase so `oo*exp(I*theta)` for some `theta`. With that interpretation I would say that a `zoo` result could possibly be equal to an `oo` or `-oo` if the phase is `0` or `pi`.

I would therefore expect the following:
```julia
In [1]: zoo.is_real                                                                                                                            
Out[1]: None

In [2]: zoo.is_positive                                                                                                                        
Out[2]: None

In [3]: zoo.is_negative                                                                                                                        
Out[3]: None
```

Does that seem reasonable?

**Comment by jksuom** [comment]
I don't think that `zoo` should be identified with `oo*exp(I*theta)`. That has currently no meaning in SymPy (except for `theta` a multiple of `pi/2`) but it could be used as an endpoint of a contour integral. `oo` and `-oo` are elements of the ordered set of extended real numbers. `zoo` is an element of the extended complex plane that has no order. There is a natural mapping taking `oo` and `-oo` to `zoo` but IMO that does not mean that the order properties should be respected.

**Comment by oscarbenjamin** [comment]
But does that not mean that `is_positive` should be `None` rather than `False`?

Where is the meaning of statements like `is_positive` (or `zoo` for that matter) defined?

**Comment by ShubhamKJha** [comment]
@oscarbenjamin please add `assumptions` label as well. It would help to sort things defining assumptions.
Personally I think `zoo` should not be real . I thought the structure of `zoo` should have been something like `oo + I*oo` which made it complex hence the above results hold.

**Comment by jksuom** [comment]
`is_positive` should mean "extended real and greater than 0". That should be `False` for (extended) complex numbers that are not real.

**Comment by oscarbenjamin** [comment]
@ShubhamKJha SymPy has such objects separately:
```
In [12]: z                                                                                                                                     
Out[12]: ∞ + ∞⋅ⅈ

In [13]: re(z)                                                                                                                                 
Out[13]: ∞

In [14]: z.is_real                                                                                                                             
Out[14]: False

In [18]: z == zoo                                                                                                                              
Out[18]: False
```

@jksuom I would like to see where all of these are defined including the implicitly assumed relationships between them. However it is the definition of zoo that I'm more interested in. Does SymPy define this anywhere?

Here is the Wolfram definition of the analogous concept used there:
[http://mathworld.wolfram.com/ComplexInfinity.html](http://mathworld.wolfram.com/ComplexInfinity.html):
```
Complex infinity is an infinite number in the complex plane
whose complex argument is unknown or undefined.
```
That matches with some properties of zoo in SymPy. For example
```
In [33]: zoo.is_complex                                                                                                                        
Out[33]: True

In [19]: abs(zoo)                                                                                                                              
Out[19]: ∞

In [20]: arg(zoo)                                                                                                                              
Out[20]: nan
```
so we know the magnitude is oo but the phase is unknown.

This also makes sense for e.g.:
```
In [24]: 1/S(0)
Out[24]: zoo
```
Stretching that last example we have
```
In [37]: x = Symbol('x', real=True)                                                                                                            

In [38]: (1/x).subs(x, 0)                                                                                                                      
Out[38]: zoo
```
where I have substituted a real value into a real expression. There is no reason that the result can't be (extended) real as it is for some limits:
```
In [39]: (1/x).limit(x, 0, '+')                                                                                                                
Out[39]: ∞

In [40]: (1/x).limit(x, 0, '-')                                                                                                                
Out[40]: -∞
```
Of course the limit has uncertain sign but there is nothing complex about it. It makes sense to use zoo to represent 1/0 because it incorporates the uncertain phase/sign. But then we should allow that zoo could mean oo or -oo as well so it is not known that zoo represents a non-real value.

**Comment by jksuom** [comment]
> There is no reason that the result can't be (extended) real as it is for some limits:

`1/0` cannot be `oo` or `-oo` as it has not definite sign. In principle, there could be yet another infinite real, say, `omega = 1/0`, with no sign, but I think that there is no reason why it could not be identified with `zoo` in the same way as real zero is identified with complex zero. It is possible to add such an element to every field, to the field of rational numbers, for example. But I see no reason why `(1/x).subs(x, 0)` should be different from `zoo` when `x = Symbol('x', rational=True)`.

**Comment by oscarbenjamin** [comment]
You seem to misunderstand my suggestion. I think it's fine that `1/0` is `zoo`.  I think though that the implications of that should be that `zoo.is_real` should be `None` since `zoo` can be the result of a purely real-valued expression.

There's nothing inconsistent about saying zoo.is_real is None if a definition like Wolfram's is used.

**Comment by jksuom** [comment]
Actually, `oo.is_real` should also be `False`. That is what the new assumptions module says (`ask(Q.real(oo))`). It is a long-term plan that `oo.is_real` be `False` while `oo.is_extended_real` be `True`. I don't think that `zoo.is_real` should be changed now as both it and `zoo.is_extended_real` are expected to be `False` in the future.

**Comment by oscarbenjamin** [comment]
That makes sense but it pushes the same problem onto is_extended_real. I think that oo.is_extended_real is True is inconsistent with zoo.is_extended_real is False (same as they are inconsistent now but the attribute is called is_real right now).

The same problem will still be there for is_positive/is_negative as well. Those should not return definite results for zoo.
